### ISSUE TITLE
Bug report for `trillium-proxy`: does not response on status 204, 304

### ISSUE THREAD

**Issue by kkazuo** [trillium-rs/trillium#79]
**Describe the bug**
Proxy does not response on status 204, 304.

**To Reproduce**
1. Run the proxy example code: https://trillium.rs/handlers/proxy.html  (it's proxy to `https://httpbin.org/`)
2. curl on 'localhost:8080/status/204'
3. Does not response

**Expected behavior**

Should response with no body.

**Screenshots**

from example code:

```rust
use trillium_logger::Logger;
use trillium_rustls::RustlsConnector;
use trillium_smol::TcpConnector;

type Proxy = trillium_proxy::Proxy<RustlsConnector<TcpConnector>>;

pub fn main() {
    env_logger::init();
    trillium_smol::run((Logger::new(), Proxy::new("https://httpbin.org/")));
}
```

with proxy:
```
% curl localhost:8080/status/204 -i
HTTP/1.1 204 No Content
access-control-allow-credentials: true
access-control-allow-origin: *
connection: keep-alive
content-type: text/html; charset=utf-8
date: Sun, 18 Jul 2021 12:40:56 GMT
server: gunicorn/19.9.0
transfer-encoding: chunked
```

without proxy:
```
% curl https://httpbin.org/status/204 -i
HTTP/2 204 
date: Sun, 18 Jul 2021 12:48:50 GMT
server: gunicorn/19.9.0
access-control-allow-origin: *
access-control-allow-credentials: true
```

**Desktop (please complete the following information):**
 - OS: macOS Big Sur 11.4
 - Browser curl
 - Version 7.64.1

**Comment by jbr** [comment]
Thanks for filing this! It makes total sense why this is happening and I'll get a patch release out tomorrow!

**Comment by jbr** [comment]
This issue is addressed by [trillium-client 0.1.1](https://github.com/trillium-rs/trillium/releases/tag/trillium-client-v0.1.1) and should be addressed by running `cargo update`. Please reopen this issue if that does not solve the problem, and thanks again for reporting this!
